Documentation/joss/paper.md

```diff
@@ -19,50 +9,84 @@ authors:
    index: 1
- - name: ??
-   index: 2
 date: 24 December 2020
 bibliography: paper.bib
 ---
 
 # Summary
 
-> TODO: A summary describing the high-level functionality and purpose of the software for a diverse, *non-specialist* audience.
+*TUM Open Infra Platform* (OIP) is an open source application for
+ viewing and analysis of different BIM models used in the civil engineering field.
+OIP supports reading, visualization, navigating, and handling of:
 
-TUM OpenInfraPlatform (OIP) is a Software-as-a-Service providing an open source application for
- viewing and analysis of different digital models used in the civil engineering field.
+- Industry Foundation Classes (IFC) models as specified in ISO 16739 [@iso16739];
+- Point Cloud Data (PCD) models as supported by the Cloud Compare library [CloudCompare].
 
-OIP supports reading and visualization of:
+Multiple models can be loaded at once and compared between each other (see Figure \ref{fig:ifcpcd}).
+Their absolute position is accounted for, so the models can be checked against one another for internal differences based on location of elements.
 
-- Industry Foundation Classes (IFC) models as specified in ISO 16739
-- Point Cloud Data (PCD) models as supported by the Cloud Compare library
+![A PCD model together with an IFC model loaded in OIP.\label{fig:ifcpcd}](../images/ifc_and_pcd.png)
 
-Multiple models can be loaded at once and compared between each other.
+Additionally, OIP incorporates its own EXPRESS lexer and parser.
+This enables automatic code generation for IFC early-binding library [@Hecht:2019:FBI].
+As such, complete contents of IFC files can be interpreted and analysed.
+Moreover, type safety is guaranteed at compile time thus reducing the risk for bugs and errors.
 
-Additionally, OIP incorporates its own EXPRESS parser to automatically generate early-binding code for IFC [Hecht:2019:FBI].
+Let it be stated here that OIP serves as a prototypical playground for developments.
+During the past 3 years, OIP was completely restructured and its code base overhauled - the IFC now stands as the internal model [@Hecht:2019:FBI].
+Major elements were already realigned, while some (previously available) functionality is still considered as work-in-progress.
 
 # Statement of Need
 
-> TODO: A clear Statement of Need that illustrates the research purpose of the software.
+The Architecture, Engineering and Construction (AEC) domain is in its transition from 2D design processes to 3D object-oriented modelling. 
+Building Information Modelling (BIM) is steadily gaining importance replacing the conventional Computer-Aided Design (CAD) practices
+ and getting implemented in every aspect of the very complex software and stakeholder landscape [@Jaud:2019:iccce].
+
+During the development of IFC standard, a clear need for an independent software
+ capable of consuming and producing IFC files according to the newly developed schemata was determined [@Amann:2015:OIP; @Amann:2015:ICCBEI].
```

Comment:
Gives a bit the impression that we are also producing IFC files
But since it is specified later on what we can do and what we can't I guess there is no need to change

Reply:
It is not that far-fetched. I believe, with a little bit of effort, we could be writing IFC files soon. there are some functions already prepared for that, e.g. `getStepLine()` gets generated for each entity.